### PR TITLE
Add support for defaultFetchSize URL parameter

### PR DESCRIPTION
**Pull request by RobertDeRose**
Adds parity with MySQL Connector/J to set fetch size globally.

### CODE REVIEW

**Comment by RobertDeRose** [comment]
Failures are due to Windows tests it appears to be related to downloading something, completely unrelated to this PR

**Comment by RobertDeRose** [comment]
@rusher What can I do to get this PR approved and merged?

File: src/main/java/org/mariadb/jdbc/MariaDbStatement.java [hunk 1]
```diff
@@ -128,6 +128,12 @@ public MariaDbStatement(MariaDbConnection connection, int resultSetScrollType, i
     this.resultSetConcurrency = resultSetConcurrency;
     this.lock = this.connection.lock;
     this.options = this.protocol.getOptions();
+    try {
+      setFetchSize(this.options.defaultFetchSize);
+    }
+    catch(SQLException e) {
+      logger.warn("defaultFetchSize is invalid, using default of 0");
```
Comment (rusher) [inline src/main/java/org/mariadb/jdbc/MariaDbStatement.java]: having a try catch and a warning for every statement has to be avoided.

"defaultFetchSize" value must be validated beforehand, then just initialize fetchSize to defaultFetchSize. 
Best seems to be in DefaultOptions.optionCoherenceValidation or something like that.

File: src/main/java/org/mariadb/jdbc/MariaDbStatement.java [hunk 2]
```diff
@@ -145,8 +151,14 @@ public MariaDbStatement clone(MariaDbConnection connection) throws CloneNotSuppo
     clone.batchQueries = new ArrayList<>();
     clone.closed = false;
     clone.warningsCleared = true;
-    clone.fetchSize = 0;
     clone.maxRows = 0;
+    try {
+      setFetchSize(this.options.defaultFetchSize);
+    }
```
Comment (rusher) [inline src/main/java/org/mariadb/jdbc/MariaDbStatement.java]: same than previous comment

**Comment by rusher** [comment]
Nice PR, @RobertDeRose 

Remains the point commented before and similar to other open source projects, the MariaDB Foundation needs to have shared ownership of all code that is included in the MariaDB distribution. The easiest way to achieve this is by submitting your code under the BSD-new license. (The other alternative is to sign the code contribution agreement which can be found here: https://mariadb.com/kb/en/mariadb/mca/)

Please indicate in a comment below that you are contributing your new code of the whole pull request under the BSD-new license or that you have filled out the contribution agreement and sent it.

Thanks,
diego

**Comment by rusher** [comment]
(and btw, https://github.com/MariaDB/mariadb-connector-j/commit/c22373d688481fbe264ff56f257107877eab244a change testing for windows, to avoid this kind of DL error).

File: src/main/java/org/mariadb/jdbc/MariaDbStatement.java (hunk 1 above)
Comment (RobertDeRose) [reply src/main/java/org/mariadb/jdbc/MariaDbStatement.java]: OK thanks, I will make the change today!

Comment (RobertDeRose) [reply src/main/java/org/mariadb/jdbc/MariaDbStatement.java]: I didn't like that either too be honest, but I didn't see that DeFaultOptions had validation. Thanks for pointing that out

**Comment by RobertDeRose** [comment]
OK, I have signed and sent in via email the MCA, I will make the requested changes hopefully today

**Comment by ColinNg** [comment]
Is this held up because of the AppVeyor Build? It seems that the mirror where it gets the MSI packages from, http://mariadb.mirrors.ovh.net/MariaDB/, is missing several versions:

10.2.2 (missing)
10.1.38 (missing)
10.3.13 ([present](http://mariadb.mirrors.ovh.net/MariaDB/mariadb-10.3.13/winx64-packages/mariadb-10.3.13-winx64.msi))
10.0.38 (present)
5.5.63 (present)

**Comment by ColinNg** [comment]
> Failures are due to Windows tests it appears to be related to downloading something, completely unrelated to this PR

It seems that the mirror where the AppVeyor Build it gets the MSI packages from, http://mariadb.mirrors.ovh.net/MariaDB/, is missing several versions:

10.2.2 (missing)
10.1.38 (missing)
10.3.13 (present)
10.0.38 (present)
5.5.63 (present)

**Comment by RobertDeRose** [comment]
@ColinNg rusher stated the issue was fix in another PR. Hopefully that just means a rebase is need for me to correct it.  I'm hoping to finish this up tonight.

**Comment by RobertDeRose** [comment]
@rusher I hope these changes remove you're concerns from the previous submission.

**Comment by RobertDeRose** [comment]
Thanks a lot!

**Pull request merged**